mapstruct-protobuf3/spi-impl/src/main/java/org/mapstruct/example/protobuf/ProtobufAccessorNamingStrategy.java

```diff
@@ -0,0 +1,33 @@
+package org.mapstruct.example.protobuf;
+
+import org.mapstruct.ap.spi.DefaultAccessorNamingStrategy;
+
+import javax.lang.model.element.Element;
+import javax.lang.model.element.ExecutableElement;
+import javax.lang.model.element.TypeElement;
+import javax.lang.model.type.DeclaredType;
+import javax.lang.model.type.TypeMirror;
+
+/**
+ * @author Thomas Kratz
+ */
+public class ProtobufAccessorNamingStrategy extends DefaultAccessorNamingStrategy {
+
+    @Override
+    public String getElementName(ExecutableElement adderMethod) {
+
+        String methodName = super.getPropertyName(adderMethod);
+        Element receiver = adderMethod.getEnclosingElement();
+        if (receiver != null) {
+            TypeElement type = (TypeElement) ((DeclaredType) receiver).asElement();
```

Comment:
That looks nicer. I was glad to make it work, have never used this part of api before.